### DISCUSSION TITLE
Request url of JSON sent by getServerSideProps has predictable path ?

### DISCUSSION THREAD

**Discussion by prasanna1211** [vercel/next.js#13382] [Help]
I was using `getServerSideProps` for my use case and everything looks so good. But I have a problem which is the permalink of the request url for JSON that is generated for each page is predictable. For example, if I have a listing page of say fruits, the permalink looks like JSON generated for each page goes like this

`example.com/_next/data/${randomnextId}/fruits/apple`
`example.com/_next/data/${randomnextId}/fruits/orange`
`example.com/_next/data/${randomnextId}/fruits/banana`

And the response of each JSON has the entire data required for that page. This would turn out easy for users to scrap the entire data when they have the all the permalinks that are possible (List of all fruits here).

Is there a way with which I can get around by blocking it ?

**Comment by tomaszrondio** [comment]
+1. 
I need to check if given resource was cached by service worker to mark it as available/unavailable when app is in offline mode.

```
caches.match(new Request('http://localhost:3000/_next/data/pWeuveh0C7M_f1BQnlmBX/opinion/your-experience-has-been-a-most-entertaining-one.json')).then(function(response) {
  console.log(response);
});
```

the problem is I don't know `randomnextId` at the level of component.

**Comment by prasanna1211** [comment]
@tomaszrondio I think you can control it through the configure the `buildId` and can be injected inside the app. [Link](https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id)

**Reply by tomaszrondio** [reply]
that looks promising. Any ideas how can I get buildId in the app? Assuming that I don't want to hard code it to fixed value.

**Reply by tomaszrondio** [reply]
ok i can also get it like that in next.config.js:
```
 webpack: (config, options) => {
    console.log('buildId', options.buildId);
    return config;
  },
```
I could write it to some file but maybe there is another/better option?
